mesonbuild/modules/sourceset.py

```diff
@@ -197,8 +197,6 @@ def add_all_method(self, state: ModuleState, args: T.Tuple[T.List[SourceSetImpl]
             raise InterpreterException('add_all called with both positional and keyword arguments')
         keys, dependencies = self.check_conditions(when)
         for s in if_true:
-            if not isinstance(s, SourceSetImpl):
-                raise InvalidCode('Arguments to \'add_all\' after the first must be source sets')
```

Comment:
Indeed unnecessary, this code predated typed_pos_args/typed_kwargs.

Reply:
Incorporated comment into commit message.